cms/middleware/toolbar.py

```diff
@@ -22,6 +23,14 @@ class ToolbarMiddleware(MiddlewareMixin):
     Middleware to set up CMS Toolbar.
     """
 
+    def is_edit_mode(self, request):
+        try:
+            match = resolve(request.path_info)
```

Comment:
why not request.resolver_match

Reply:
during this time request.resolver_match is None. the same implementation is done in is_cms_request in the ToolbarMiddleware.

Reply:
In the Middleware the resolver_match is always null in the Middleware. This is per Djangos docs:

https://docs.djangoproject.com/en/2.1/ref/request-response/#django.http.HttpRequest.resolver_match